community/kernel/src/main/java/org/neo4j/GraphDatabaseBuilder.java

```diff
@@ -0,0 +1,384 @@
+     * Set the threshold, in bytes, after which logical logs will be pruned.
+     *
+     * @param thresholdSize the threshold, in bytes, after which logical logs will be pruned
+     * @return this builder
+     */
+    public BUILDER withLogicalLogPruningThresholdSize( long thresholdSize )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( thresholdSize / 1024 ) + " k" );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of archived logical logs.
+     *
+     * @param count the maximum number of archived logical logs
+     * @return this builder
+     */
+    public BUILDER withLogicalLogMaxArchives( int count )
+    {
+        params.put( GraphDatabaseSettings.keep_logical_logs.name(), String.valueOf( count ) + " files" );
+        return self();
+    }
+
+    /**
+     * Set the threshold, in bytes, for logical log rotation.
+     *
+     * @param threshold the threshold, in bytes, for logical log rotation
+     * @return this builder
+     */
+    public BUILDER withLogicalLogRotationThreshold( long threshold )
+    {
+        params.put( GraphDatabaseSettings.logical_log_rotation_threshold.name(), String.valueOf( threshold ) );
+        return self();
+    }
+
+    /**
+     * The amount of memory to use for mapping the store files, in bytes. If Neo4j is running on a dedicated server,
+     * then it is generally recommended to leave about 2-4 gigabytes for the operating system, give the
+     * JVM enough heap to hold all your transaction state and query context, and then leave the rest for
+     * the page cache. If this is not set, then it is assumed that Neo4j is running exclusively on the machine,
+     * and the size will be heuristically set to 75% of RAM minus the max Java heap size.
+     *
+     * @param size the amount of memory to use for mapping the store files, in bytes
+     * @return this builder
+     */
+    public BUILDER usingPageCacheMemory( long size )
+    {
+        params.put( GraphDatabaseSettings.pagecache_memory.name(), String.valueOf( size ) );
+        return self();
+    }
+
+    /**
+     * Set an internal parameter. Use of this method is not recommended.
+     *
+     * @param key parameter key
+     * @param value parameter value
+     * @return this builder
+     */
+    public BUILDER withParam( String key, String value )
+    {
+        this.params.put( key, value );
+        return self();
+    }
+
+    /**
+     * Set internal parameters. Use of this method is not recommended.
+     *
+     * @param params parameters to add
+     * @return this builder
+     */
+    public BUILDER withParams( Map<String,String> params )
+    {
+        this.params.putAll( params );
+        return self();
+    }
+
+    /**
+     * Starts the graph database.
+     *
+     * @param storeDir The filesystem location for the store, which will be created if necessary
+     * @return The running database
+     */
+    public GRAPHDB open( File storeDir )
```

Comment:
I'd prefer not to have `String` - it's not the responsibility of this library to figure out how to parse a string into a File / Path. I totally agree on having `Path`, but I'd want the code to respect the filesystem attached to the `Path` object, and not throw it away by converting to `File` internally: so I thought it might have to wait until you and @chrisvest have finished changing everything to use `Path` internally?